ironfish/src/mining/manager.ts

```diff
@@ -184,8 +185,8 @@ export class MiningManager {
 
     const blockDisplay = `${block.header.hash.toString('hex')} (${block.header.sequence})`
     if (
-      !this.node.chain.head ||
-      !block.header.previousBlockHash.equals(this.node.chain.head.hash)
+      !block.header.previousBlockHash.equals(this.node.chain.head.hash) &&
+      !isBlockHeavier(block.header, this.node.chain.head)
```

Comment:
Two unrelated comments:
1. Since work should be strictly increasing, I think we could even skip the previousBlockHash equality check and just toss out all blocks that aren't heavier than the chain head.
2. This change will let through blocks that are heavier, but don't attach to the chain head. I don't think this should be common -- seems like it should only happen in edge cases like rolling back a multi-block fork. Regardless, I don't think this should be an issue, since `addBlock` will drop the block if it doesn't connect anyway.

Reply:
I think we may be able to remove this check and the `CHAIN_CHANGED` case entirely.

We already check `isBlockHeavier` in the `addBlock` logic. The only benefit to returning early here is to avoid the operations for verifying the block and adding it to a fork.

Reply:
I think it may be better to leave `CHAIN_CHANGED` in for now.

We still need to make a change here to compute the work for the submitted block from its previous block or `isBlockHeavier` will always be false